controllers/programmes/index.js

```diff
@@ -0,0 +1,310 @@
+            .filter(programmeFilters.filterByMinAmount(minAmountParam))
+            .filter(programmeFilters.filterByMaxAmount(maxAmountParam));
+
+        templateData.activeBreadcrumbs.push({
+            label: globalCopy.nav.funding,
+            url: req.baseUrl
+        });
+
+        if (!minAmountParam && !maxAmountParam && !locationParam) {
+            templateData.activeBreadcrumbs.push({
+                label: copy.breadcrumbAll
+            });
+        } else {
+            templateData.activeBreadcrumbs.push({
+                label: copy.title,
+                url: req.baseUrl + req.path
+            });
+
+            if (parseInt(minAmountParam, 10) === 10000) {
+                templateData.activeBreadcrumbs.push({
+                    label: copy.over10k,
+                    url: '/over10k'
+                });
+            }
+
+            if (parseInt(maxAmountParam, 10) === 10000) {
+                templateData.activeBreadcrumbs.push({
+                    label: copy.under10k,
+                    url: '/under10k'
+                });
+            }
+
+            if (locationParam) {
+                const locationParamToTranslation = key => {
+                    const regions = {
+                        england: globalCopy.regions.england,
+                        wales: globalCopy.regions.wales,
+                        scotland: globalCopy.regions.scotland,
+                        northernIreland: globalCopy.regions.northernIreland,
+                        ukWide: globalCopy.regions.ukWide
+                    };
+                    return regions[key];
+                };
+
+                templateData.activeBreadcrumbs.push({
+                    label: locationParamToTranslation(locationParam),
+                    count: templateData.programmes.length
+                });
+            }
+        }
+
+        templateData.activeBreadcrumbsSummary = map(templateData.activeBreadcrumbs, 'label').join(', ');
+
+        res.render(path.resolve(__dirname, './views/programmes-list'), templateData);
+    });
+}
+
+/**
+ * Route: Programme Detail
+ */
+function initProgrammeDetail(router) {
+    router.get('/programmes/:slug', injectFundingProgramme, (req, res, next) => {
+        const entry = res.locals.fundingProgramme;
+
+        if (entry && entry.contentSections.length > 0) {
+            res.render(path.resolve(__dirname, './views/programme'), {
+                entry: entry,
+                title: entry.summary.title,
+                heroImage: entry.hero || heroImages.fallbackHeroImage,
+                isBilingual: isBilingual(entry.availableLanguages)
+            });
+        } else {
+            next();
+        }
+    });
+}
+
+/**
+ * Route Strategic Programme Detail
+ */
+function initStrategicProgrammeDetail(router) {
+    router.get('/strategic/headstart', function(req, res) {
+        // Mock entry
+        const entry = {
```

Comment:
Could maybe store this mock data in a JSON file somewhere (in tests folder?) rather than having it in the history of this file etc?